src/webui/src/components/Modal/CustomizedTrial.tsx

```diff
@@ -0,0 +1,329 @@
+            // open the warning modal
+            if (this._isCustomizeMount === true) {
+                this.setState(() => ({ isShowWarning: true, customParameters: customized }));
+            }
+        } else {
+            // submit a customized job
+            this.submitCustomize(customized);
+        }
+
+    }
+
+    warningConfirm = () => {
+        if (this._isCustomizeMount === true) {
+            this.setState(() => ({ isShowWarning: false }));
+        }
+        const { customParameters } = this.state;
+        this.submitCustomize(customParameters);
+    }
+
+    warningCancel = () => {
+        if (this._isCustomizeMount === true) {
+            this.setState(() => ({ isShowWarning: false }));
+        }
+    }
+
+    submitCustomize = (customized: Object) => {
+        // delete `tag` key
+        for (let i in customized) {
+            if (i === 'tag') {
+                delete customized[i];
+            }
+        }
+        axios(`${MANAGER_IP}/trial-jobs`, {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            data: customized
+        })
+            .then(res => {
+                if (res.status === 200) {
+                    if (this._isCustomizeMount === true) {
+                        this.setState(() => ({ customID: res.data.sequenceId , isShowSubmitSucceed: true }));
+                        this.props.closeCustomizeModal();
+                    }
+                } else {
+                    if (this._isCustomizeMount === true) {
+                        this.setState(() => ({ isShowSubmitFailed: true }));
+                    }
+                }
+            })
+            .catch(error => {
+                if (this._isCustomizeMount === true) {
+                    this.setState(() => ({ isShowSubmitFailed: true }));
+                }
+            });
+    }
+
+    closeSucceedHint = () => {
+        // also close customized trial modal
+        if (this._isCustomizeMount === true) {
+            this.setState(() => ({ isShowSubmitSucceed: false }));
+            this.props.closeCustomizeModal();
+        }
+    }
+
+    closeFailedHint = () => {
+        // also close customized trial modal
+        if (this._isCustomizeMount === true) {
+            this.setState(() => ({ isShowSubmitFailed: false }));
+            this.props.closeCustomizeModal();
+        }
+    }
+
+    componentDidMount() {
+        this._isCustomizeMount = true;
+    }
+
+    componentWillUnmount() {
+        this._isCustomizeMount = false;
+    }
+
+    render() {
+        const { visible, closeCustomizeModal, hyperParameter } = this.props;
+        const { isShowSubmitSucceed, isShowSubmitFailed, isShowWarning, customID } = this.state;
+        const {
+            form: { getFieldDecorator },
+            // form: { getFieldDecorator, getFieldValue },
+        } = this.props;
+        const parameters = (hyperParameter !== '') ? JSON.parse(hyperParameter) : '';
```

Comment:
If your purpose is to mark empty hyper-parameters, `null` is better than `''`.

Reply:
Ok.